types/array-binarysearch.closest/index.d.ts

```diff
@@ -0,0 +1,13 @@
+// Type definitions for array-binarysearch.closest 0.2
+// Project: https://www.npmjs.com/package/array-binarysearch.closest
+// Definitions by: Heye Vöcking <https://github.com/hvoecking>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export function binarySearch<T>(
+    arr: ReadonlyArray<T>,
+    val: T,
+    fn?: null | ((itm: T, val: T, m: any, arr: ReadonlyArray<T>) => number),
```

Comment:
Technically this function should only be generic when `fn` is provided. Otherwise it really only safely works on `ReadonlyArray<string>`s and `ReadonlyArray<number>`s.  I'd probably make overloads to model this.